presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -335,6 +345,27 @@ protected Boolean visitFunctionCall(FunctionCall node, Void context)
                             PERFORMANCE_WARNING,
                             "COUNT(DISTINCT xxx) can be a very expensive operation when the cardinality is high for xxx. In most scenarios, using approx_distinct instead would be enough"));
                 }
+                boolean nodeIsApproximateCountDistinctFunction = functionResolution.isApproximateCountDistinctFunction(analysis.getFunctionHandle(node));
+                boolean nodeIsApproximateSetFunction = functionResolution.isApproximateSetFunction(analysis.getFunctionHandle(node));
+                if (nodeIsApproximateCountDistinctFunction || nodeIsApproximateSetFunction) {
+                    String functionName = metadata.getFunctionAndTypeManager().getFunctionMetadata(analysis.getFunctionHandle(node)).getName().getObjectName();
+                    double maxStandardError = DefaultApproximateCountDistinctAggregation.DEFAULT_STANDARD_ERROR;
+                    double lowestMaxStandardError = SystemSessionProperties.getLowestApproximateCountDistinctMaxStandardError(session);
+                    if (nodeIsApproximateSetFunction) {
+                        maxStandardError = ApproximateSetAggregation.DEFAULT_STANDARD_ERROR;
+                        lowestMaxStandardError = SystemSessionProperties.getLowestApproximateSetMaxStandardError(session);
+                    }
+                    // maxStandardError is supplied
+                    if (node.getArguments().size() > 1) {
+                        Expression maxStandardErrorExpr = node.getArguments().get(1);
+                        if (maxStandardErrorExpr instanceof DoubleLiteral) {
+                            maxStandardError = ((DoubleLiteral) maxStandardErrorExpr).getValue();
+                        }
+                    }
+                    if (maxStandardError <= lowestMaxStandardError) {
+                        warningCollector.add(new PrestoWarning(PERFORMANCE_WARNING, String.format("%s can produce low-precision results with the current standard error: %.4f (<=%.4f)", functionName, maxStandardError, lowestMaxStandardError)));
+                    }
```

Comment:
It would be good to separate into two functions. Because if a query contains both approx_distinct and approx_set, we should emit two warnings.